server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -43,13 +44,16 @@
 
 import static org.elasticsearch.core.Strings.format;
 
+/**
+ * Base class to send a request out to a list of nodes.
+ */
 public abstract class TransportNodesAction<
     NodesRequest extends BaseNodesRequest<NodesRequest>,
     NodesResponse extends BaseNodesResponse<?>,
     NodeRequest extends TransportRequest,
     NodeResponse extends BaseNodeResponse> extends TransportAction<NodesRequest, NodesResponse> {
 
-    private static final Logger logger = LogManager.getLogger(TransportNodesAction.class);
+    public static final Logger logger = LogManager.getLogger(TransportNodesAction.class);
```

Comment:
I think it is better to avoid loggers in abstract classes. In most cases once it is used it is not clear what implementation actually caused the log entry. Is there a chance instead of making it public we could remove it and replace log usages iwth concrete implementation loggers?

Reply:
FWIW I disagree with this. I find it more useful for the logger to refer to the exact source file that contains the log message being emitted, rather than having to chase messages around between classes. It's certainly important for the message to describe the context in which it's being called, but typically that means much more than just the name of the concrete subclass.

However I don't think we want this to be public. It shouldn't be needed outside of this class anyway (by the same reasoning) but if there's a good reason for needing it elsewhere then we should just get it again from the `LogManager` rather than from here.